### ISSUE TITLE
XS_PowerLaw fits differently from PowerLaw

### ISSUE THREAD

**Issue by fjebaker** [fjebaker/SpectralFitting.jl#41]
```julia
model = PhotoelectricAbsorption() * PowerLaw()

prob = FittingProblem(model, data)
result = fit(prob, LevenbergMarquadt(λ_inc = 2.0))
```
Output:
```
FittingResult:
  Model: CompositeModel
    - Parameters  : [0.009039957255391456, 1.5898093590630118, 0.0]
    - χ2          : 173569.71735048556 
```
Produces different results from
```julia
model = PhotoelectricAbsorption() * XS_PowerLaw()

prob = FittingProblem(model, data)
result = fit(prob, LevenbergMarquadt(λ_inc = 2.0))
```
Output:
```
 FittingResult:
  Model: CompositeModel
    - Parameters  : [0.011511508412245715, 1.8436247594879935, 0.0]
    - χ2          : 63271.22443456664 
```

It is likely a Dual number is being erased in the forward diff, and some gradient information is pulling the model in the wrong direction, but I am unsure.

**Comment by phajy** [comment]
@fjebaker, can you try re-running this with `model = PhotoelectricAbsorption() * XS_PowerLaw(a = FitParam(2.0))` for the second fit, please?

I think the problem might be that the Julia powerlaw and XSPEC powerlaw have different default photon indices. There's quite a big degeneracy between column density and photon index. Even though these fits both end up with a column density of zero, the $\chi^2$ space gets quite flat, and the optimiser stops at different places given different starting points. I explored this with a NuSTAR dataset that I have, and found that I get good agreement between the two models when they both start from a photon index of 2 (see output below), but produce different results if they start with their default photon indices.

```
============================================================
phabs × power-law fit comparison (3–30 keV), both start a=2
============================================================
                 Julia PowerLaw          XSPEC XS_PowerLaw
  ηH                     0.0 ± 0.0528               0.0 ± 0.0528
  K (norm)            40.034 ± 0.4644            40.034 ± 0.4644
  a (index)          3.76369 ± 0.005575           3.76369 ± 0.005575
  χ²                 81441.2                         81441.2
------------------------------------------------------------
```

I found that both models produce the same $\chi^2$ value when they're evaluated with the same parameters.

The bottom line is that I don't think this is a "bug". Perhaps if the optimiser tolerance was made smaller they would both converge on the same minimum. Alternatively, SpectralFitting has changed sufficiently since this issue was raised that any bug may have fixed itself!

If the test produces the same result for you, this issue can be closed.
